showcase/app/templates/page-components/copy/button.gts

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+import Component from '@glimmer/component';
+import { array } from '@ember/helper';
+import { capitalize } from '@ember/string';
+import { eq } from 'ember-truth-helpers';
+import { get } from '@ember/object';
+import { pageTitle } from 'ember-page-title';
+import style from 'ember-style-modifier';
+
+import ShwDivider from 'showcase/components/shw/divider';
+import ShwFlex from 'showcase/components/shw/flex';
+import ShwGrid from 'showcase/components/shw/grid';
+import ShwOutliner from 'showcase/components/shw/outliner';
+import ShwTextH1 from 'showcase/components/shw/text/h1';
+import ShwTextH2 from 'showcase/components/shw/text/h2';
+import ShwTextH3 from 'showcase/components/shw/text/h3';
+import ShwTextH4 from 'showcase/components/shw/text/h4';
+
+import { HdsCopyButton } from '@hashicorp/design-system-components/components';
+import { SIZES } from '@hashicorp/design-system-components/components/hds/copy/button/index';
+
+import CopyButtonStates from 'showcase/components/mock/components/copy/button/copy-button-states';
+import CopyButtonWithMaskedInput, {
+  MASKED_INPUT_VARIANTS,
+} from 'showcase/components/mock/components/copy/button/copy-button-with-masked-input';
+import type { CopyButtonWithMaskedInputSignature } from 'showcase/components/mock/components/copy/button/copy-button-with-masked-input';
+import CopyButtonWithInput, {
+  INPUT_COMPONENTS,
+} from 'showcase/components/mock/components/copy/button/copy-button-with-input';
+import CopyButtonInModal from 'showcase/components/mock/components/copy/button/copy-button-in-modal';
+import CopyButtonInDropdown from 'showcase/components/mock/components/copy/button/copy-button-in-dropdown';
+import CopyButtonWithVanillaInput, {
+  INPUT_COMPONENTS as VANILLA_INPUT_COMPONENTS,
+} from 'showcase/components/mock/components/copy/button/copy-button-with-vanilla-input';
+
+const maskedInputVariantToLabelMap: Record<
+  CopyButtonWithMaskedInputSignature['Args']['variant'],
+  string
+> = {
+  'masked-input-base': 'With MaskedInput::Base',
+  'masked-input-base-form-field': 'With Form::Field + MaskedInput::Base',
+  'masked-input-field': 'With MaskedInput::Field',
+};
+
+export default class PageComponentCopyButton extends Component {
+  get bigIntNumber() {
+    const bigIntNumber = BigInt(12345678910);
+    return bigIntNumber;
+  }
+
+  get targetNodeElement() {
+    const element = document.querySelector('#test-target-node-element');
```

Comment:
you can probably just do this in the constructor and have `#element` class member

Reply:
I tried that out but `#element` didnt exist

Reply:
yeah you need declare it manually first